dex/pair/src/safe_price_view.rs

```diff
@@ -472,10 +513,76 @@ pub trait SafePriceViewModule:
             second_token_reserve_accumulated,
             weight_accumulated,
             recording_round: search_round,
+            recording_timestamp: interpolated_timestamp,
             lp_supply_accumulated,
         }
     }
 
+    fn find_observation_by_timestamp(
+        &self,
+        target_timestamp: u64,
+        current_index: usize,
+        price_observations: &VecMapper<Self::Api, PriceObservation<Self::Api>, ManagedAddress>,
```

Comment:
I don't think you have to specify `Self::Api` here.

```suggestion
        price_observations: &VecMapper<PriceObservation<Self::Api>, ManagedAddress>,
```

Reply:
Done.